src/serialize/value.rs

```diff
@@ -0,0 +1,906 @@
+        let expected = match self.map_type_to_expected() {
+            Ok(expected) => expected,
+            Err(e) => return e,
+        };
+        SerializationError::new(mk_typck_err_named(
+            name,
+            typ,
+            BuiltinTypeCheckErrorKind::MismatchedType { expected },
+        ))
+    }
+
+    fn serialize_natives<'a>(
+        &self,
+        typ: &ColumnType,
+        native_type: &NativeType,
+        cell_writer: CellWriter<'a>,
+    ) -> Result<WrittenCellProof<'a>, SerializationError> {
+        match native_type {
+            NativeType::Ascii => self.serialize_text(typ, cell_writer),
+            NativeType::Boolean => self.serialize_native::<bool>(typ, cell_writer),
+            NativeType::Blob => self.serialize_native::<Vec<u8>>(typ, cell_writer),
+            NativeType::Counter => {
+                let value = self.extract::<i64>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<Counter>(), typ)
+                })?;
+
+                let counter = Counter(value);
+                counter.serialize(typ, cell_writer)
+            }
+            NativeType::Date => {
+                let value = self.extract::<i64>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlDate>(), typ)
+                })?;
+
+                let value: u32 = u32::try_from(value).map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlDate>(), typ)
+                })?;
+
+                let date = CqlDate(value);
+                date.serialize(typ, cell_writer)
+            }
+            NativeType::Decimal => {
+                let value = self.extract::<(Vec<u8>, i32)>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlDecimal>(), typ)
+                })?;
+
+                let decimal = CqlDecimal::from_signed_be_bytes_and_exponent(value.0, value.1);
+                decimal.serialize(typ, cell_writer)
+            }
+            NativeType::Double => self.serialize_native::<f64>(typ, cell_writer),
+            NativeType::Duration => {
+                let value = self.extract::<(i32, i32, i64)>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlDuration>(), typ)
+                })?;
+
+                let duration = CqlDuration {
+                    months: value.0,
+                    days: value.1,
+                    nanoseconds: value.2,
+                };
+                duration.serialize(typ, cell_writer)
+            }
+            NativeType::Float => self.serialize_native::<f32>(typ, cell_writer),
+            NativeType::Int => self.serialize_native::<i32>(typ, cell_writer),
+            NativeType::BigInt => self.serialize_native::<i64>(typ, cell_writer),
+            NativeType::Text => self.serialize_text(typ, cell_writer),
```

Comment:
♻️ Can you please group `NativeType` by categories? I.e., int types ordered by ascending size, then floats ordered by ascending size, then boolean, then...